openfisca_core/taxscales.py

```diff
@@ -209,21 +256,46 @@ def add_tax_scale(self, tax_scale):
                 self.combine_bracket(rate, threshold_low, threshold_high)
             self.combine_bracket(tax_scale.rates[-1], tax_scale.thresholds[-1])  # Pour traiter le dernier threshold
 
-    def calc(self, base, factor = 1, round_base_decimals = None):
-        base1 = np.tile(base, (len(self.thresholds), 1)).T
-        if isinstance(factor, (float, int)):
-            factor = np.ones(len(base)) * factor
-        # np.finfo(np.float).eps is used to avoid np.nan = 0 * np.inf creation
-        thresholds1 = np.outer(factor + np.finfo(np.float).eps, np.array(self.thresholds + [np.inf]))
+    def calc(
+            self,
+            tax_bases: ndarray,
```

Comment:
May be we should stick to (singular) tax_base. You might precise that it is a vector in the description.

Reply:
Sure, sorry I just moved things around so to understand the behaviour.